--- prometheus.rb.ORIG	2019-06-07 16:42:24.740298352 +0200
+++ prometheus.rb	2019-06-07 16:44:04.235235930 +0200
@@ -34,6 +34,7 @@
       parse_prometheus_alertmanager_config
       parse_alertmanager_config
       parse_scrape_configs
+      relabel_scrape_configs
       parse_rules_files
       parse_flags
     end
@@ -260,6 +261,19 @@
       prometheus_scrape_configs
     end
 
+    def relabel_scrape_configs
+      Gitlab['prometheus']['scrape_configs'] = Gitlab['prometheus']['scrape_configs'].map do |c|
+        c['static_configs'] = c['static_configs'].map do |s|
+          s['labels'] ||= {}
+          s['labels']['instance'] = "docker-" + `hostname`.chomp
+          s['labels']['service'] = "gitlab"
+          s['labels']['environment'] = ENV['ENVIRONMENT']
+          s
+        end
+        c
+      end
+    end
+
     def gitaly_scrape_config
       # Don't parse if gitaly is explicitly disabled
       return unless Services.enabled?('gitaly')
